### ISSUE TITLE
Adding flags for API functions

### ISSUE THREAD

**Issue by jw-mcgrath** [HoloClean/HoloClean-Legacy-deprecated#41]
We should have flags (with defaults) for things like

- verbose/debug mode
- learning rates
- training iterations
- batch size for neural net
- weight decay
- momentum
- domain pruning threshold

Essentially, any parameter to the model for learning should be at the surface level

**Comment by JVonderwell** [comment]
I might suggest we split this into 2 issues, one for verbose and one for the functionality. Fairly different tasks.

**Comment by jw-mcgrath** [comment]
I've fixed up the verbose/debug modes, there were just some missing if statements. Still have to expose the learning parameters.
